api/census.go

```diff
@@ -102,22 +90,31 @@ func (a *API) addParticipantsHandler(w http.ResponseWriter, r *http.Request) {
 		ErrGenericInternalServerError.WithErr(err).Write(w)
 		return
 	}
-
 	// check the user has the necessary permissions
 	if !user.HasRoleFor(census.OrgAddress, db.ManagerRole) && !user.HasRoleFor(census.OrgAddress, db.AdminRole) {
 		ErrUnauthorized.Withf("user is not admin of organization").Write(w)
 		return
 	}
-
-	// add the org participants where necessary
-	no, err := a.db.SetBulkCensusMembership(passwordSalt, censusID, participantsInfo.Participants)
+	// decode the participants from the request body
+	participants := &AddParticipantsRequest{}
+	if err := json.NewDecoder(r.Body).Decode(participants); err != nil {
```

Comment:
Why do you do the decoding after check if the user has permissions?
Firsxt the requiest validity should be checked

Reply:
Now, if the user does not have permission, the code stops and returns the error. With the previous code, the decoding is done regardless of whether the user meets the role requirements or not.

Reply:
So if there is a problematic argument why not avoid the user check?
In general i do no think what you propose is a good practice. Though considering the possible size of this request I think it can make sense